From patchwork Wed Aug  1 13:30:22 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peng Hao <peng.hao2@zte.com.cn>
X-Patchwork-Id: 10551705
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 8AE9B174A
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed,  1 Aug 2018 05:14:37 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7B776203B9
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed,  1 Aug 2018 05:14:37 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 6FB3F2AE2E; Wed,  1 Aug 2018 05:14:37 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No,
 score=-6.0 required=2.0 tests=BAYES_00,DATE_IN_FUTURE_06_12,
	MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id CEE1D203B9
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed,  1 Aug 2018 05:14:33 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1733038AbeHAG6R (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 1 Aug 2018 02:58:17 -0400
Received: from out1.zte.com.cn ([202.103.147.172]:56404 "EHLO mxct.zte.com.cn"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1733020AbeHAG6Q (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 1 Aug 2018 02:58:16 -0400
Received: from mse01.zte.com.cn (unknown [10.30.3.20])
        by Forcepoint Email with ESMTPS id 227893506EA671771C9A;
        Wed,  1 Aug 2018 13:14:27 +0800 (CST)
Received: from notes_smtp.zte.com.cn ([10.30.1.239])
        by mse01.zte.com.cn with ESMTP id w715EKcG077830;
        Wed, 1 Aug 2018 13:14:20 +0800 (GMT-8)
        (envelope-from peng.hao2@zte.com.cn)
Received: from localhost.localdomain.localdomain ([10.74.120.59])
          by szsmtp06.zte.com.cn (Lotus Domino Release 8.5.3FP6)
          with ESMTP id 2018080113142553-2805432 ;
          Wed, 1 Aug 2018 13:14:25 +0800
From: Peng Hao <peng.hao2@zte.com.cn>
To: pbonzini@redhat.com, rkrcmar@redhat.com, tglx@linutronix.de,
        x86@kernel.org, peterz@infradead.org, mingo@redhat.com
Cc: kvm@vger.kernel.org, zhong.weidong@zte.com.cn,
        Peng Hao <peng.hao2@zte.com.cn>
Subject: [PATCH] kvm/x86 : simplify the handle of invvpid function
Date: Wed, 1 Aug 2018 21:30:22 +0800
Message-Id: <1533130222-114621-1-git-send-email-peng.hao2@zte.com.cn>
X-Mailer: git-send-email 1.8.3.1
X-MIMETrack: Itemize by SMTP Server on SZSMTP06/server/zte_ltd(Release
 8.5.3FP6|November
 21, 2013) at 2018-08-01 13:14:25,
        Serialize by Router on notes_smtp/zte_ltd(Release 9.0.1FP7|August  17,
 2016) at
 2018-08-01 13:14:12,
        Serialize complete at 2018-08-01 13:14:12
X-MAIL: mse01.zte.com.cn w715EKcG077830
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

When vpids are not supported or disabled, vpid_sync_context is just
a no-op including single-context type or global-context type. we
can just return in vpid_sync_context when vpid==0.

Signed-off-by: Peng Hao <peng.hao2@zte.com.cn>
---
 arch/x86/kvm/vmx.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 1689f43..b76a18d 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -1968,10 +1968,7 @@ static void loaded_vmcs_clear(struct loaded_vmcs *loaded_vmcs)
 
 static inline void vpid_sync_vcpu_single(int vpid)
 {
-	if (vpid == 0)
-		return;
-
-	if (cpu_has_vmx_invvpid_single())
+	if (cpu_has_vmx_invvpid_single())
 		__invvpid(VMX_VPID_EXTENT_SINGLE_CONTEXT, vpid, 0);
 }
 
@@ -1983,6 +1980,9 @@ static inline void vpid_sync_vcpu_global(void)
 
 static inline void vpid_sync_context(int vpid)
 {
+	if (vpid == 0)
+		return;
+
 	if (cpu_has_vmx_invvpid_single())
 		vpid_sync_vcpu_single(vpid);
 	else
